ui/apps/platform/src/routePaths.ts

```diff
@@ -73,89 +73,122 @@ export const vulnManagementImagesPath = `${vulnManagementPath}/images`;
 export const vulnManagementNamespacesPath = `${vulnManagementPath}/namespaces`;
 export const vulnManagementNodesPath = `${vulnManagementPath}/nodes`;
 
+// Compose resourceAccessRequirements from resource names and predicates.
+
+type ResourcePredicate = (hasReadAccess: HasReadAccess) => boolean;
+
+type ResourceItem = ResourceName | ResourcePredicate;
+
+function evaluateItem(resourceItem: ResourceItem, hasReadAccess: HasReadAccess) {
+    if (typeof resourceItem === 'function') {
+        return resourceItem(hasReadAccess);
+    }
+
+    return hasReadAccess(resourceItem);
+}
+
+export function everyResource(resourceItems: ResourceItem[]): ResourcePredicate {
+    return (hasReadAccess: HasReadAccess) =>
+        resourceItems.every((resourceItem) => evaluateItem(resourceItem, hasReadAccess));
+}
+
+export function someResource(resourceNames: ResourceName[]): ResourcePredicate {
+    return (hasReadAccess: HasReadAccess) =>
+        resourceNames.some((resourceItem) => evaluateItem(resourceItem, hasReadAccess));
+}
+
 // Source of truth for conditional rendering of Body route paths and NavigationSidebar links.
 
 type RouteDescription = {
     featureFlagDependency?: FeatureFlagEnvVar[]; // assume multiple feature flags imply all must be enabled
-    resourceAccessRequirements: ResourceName[]; // assume READ_ACCESS and multiple resource names imply must have access to all
+    resourceAccessRequirements: ResourcePredicate; // assume READ_ACCESS
```

Comment:
Why does the new type not need an array suffix `[]` ?

Reply:
Good question. Which lead my eyes to an inconsistency above, which I will fix.

TypeScript had to remind me to delete `[]` when I edited. Now `everyResource` and `someResource`higher order functions have `ResourceItem[]` as argument and return a predicate function which receives the dependency injection as you observed above.